src/mavoscript.js

```diff
@@ -533,6 +533,51 @@ var _ = Mavo.Script = {
 
 		return [a, b];
 	},
+	/**
+	 * Returns a list of all variables in the AST
+	 */
+	getVariables: function(ast) {
+		switch (ast.type) {
+			case "Literal":
+				return [];
+			case "Identifier":
+				const isGlobal = !!globalThis[ast.name];
+				return isGlobal? [] : [ast];
+			case "UnaryExpression":
+				return _.getVariables(ast.argument);
+			case "BinaryExpression":
+				// handles group(key: value)
+				if (ast.operator === ':') {
+					return _.getVariables(ast.right);
+				}
+				return [ast.left, ast.right].flatMap(_.getVariables);
+			case "LogicalExpression":
+				return [ast.left, ast.right].flatMap(_.getVariables);
+			case "CallExpression":
+				let toExplore = [...ast.arguments];
+				// prevents adding the function name to the list of vars
+				if (ast.callee.type !== "Identifier") {
+					toExplore = [ast.callee].concat(toExplore);
+				}
```

Comment:
Nit, but I found this code a little hard to read.

If you want to add an element to the beginning of an array, you can use [`Array#unshift()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift)

But it may be better to add things in the right order to begin with, rather than having to change it later, e.g. like this:
```suggestion
				// For plain function calls (e.g. foo()), the callee is not a variable
				let toExplore = ast.callee.type === "Identifier" ? [] : [ast.callee];
				toExplore.push(...ast.arguments);
```